### PR TITLE
ftp: add options to enable TLS - fixes 3640

### LINKED ISSUE #3640: Feature request: enable serve ftp with TLS
hi, it would be useful to enable tls when using rclone as an ftp server.

rclone --ftp-tls serve ftp \
--client-cert ~/rclone/server.crt \
--client-key ~/rclone/server.key \
--auth-proxy ~/rclone/auth-proxy \
--no-seek

but if i try to connect using tls:

curl -vu test:test --ssl -l ftp://localhost:2121/
* About to connect() to localhost port 2121 (#0)
* Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 2121 (#0)
< 220 Welcome to Rclone v1.49.5 FTP Server
> > AUTH SSL
< 550 Action not taken
> > AUTH TLS
< 550 Action not taken

thnx a lot.

### PR DESCRIPTION
**Pull request by goku321**
#### What is the purpose of this change?

Enable TLS on serve ftp command
#### Was the change discussed in an issue or in the forum before?
Yes

Fixes #3640 
#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: cmd/serve/ftp/ftp.go [hunk 1]
```diff
@@ -60,6 +62,8 @@ func AddFlags(flagSet *pflag.FlagSet) {
 	flags.StringVarP(flagSet, &Opt.PassivePorts, "passive-port", "", Opt.PassivePorts, "Passive port range to use.")
 	flags.StringVarP(flagSet, &Opt.BasicUser, "user", "", Opt.BasicUser, "User name for authentication.")
 	flags.StringVarP(flagSet, &Opt.BasicPass, "pass", "", Opt.BasicPass, "Password for authentication. (empty value allow every password)")
+	flags.StringVarP(flagSet, &Opt.TLSCert, "client-cert", "", Opt.TLSCert, "TLS PEM key (concatenation of certificate and CA certificate)")
```
Comment (ncw) [inline cmd/serve/ftp/ftp.go:65]: I think these options should be called this to match the options available in `rclone serve http`

      --cert string                            SSL PEM key (concatenation of certificate and CA certificate)
      --key string                             SSL PEM Private key

If you copy the options from here then we can keep them consistent

https://github.com/rclone/rclone/blob/095c7bd801a8ef01bb4a9c792340fa3c0a302fe8/cmd/serve/httplib/httpflags/httpflags.go#L22-L23

**Review by ncw — COMMENTED** [review]
Nice feature - thanks :-)

A little consistency thing inline.

File: cmd/serve/ftp/ftp.go (hunk 1 above)
Comment (goku321) [reply cmd/serve/ftp/ftp.go:65]: Sure

Comment (goku321) [reply cmd/serve/ftp/ftp.go]: @ncw I've made the requested changes.

**Comment by ncw** [comment]
This looks great now, I'll merge it, thank you :-)

**Pull request merged**